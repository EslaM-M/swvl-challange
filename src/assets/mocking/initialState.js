// import route from "../data/route";
import tripInformation from "../data/trip-information";
export const initialState = {
  bookings: [],
  bookingAllowed: true,
  route: tripInformation.route.map(e => {
    e.count = 0;
    return e;
  }),
  captainLocation: {
    lat: null,
    lng: null
  },
  tripStatus: "NOT_STARTED",
  tripStartTime: null,
  arrivalStatus: "",
  currentStationId: null,
  fullDistance: null,
  baseFare: tripInformation.baseFare,
  startPoint: tripInformation.startPoint,
  endPoint: tripInformation.endPoint,
  driver: tripInformation.driver
};
