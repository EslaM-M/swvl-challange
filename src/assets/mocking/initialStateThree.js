// import route from "../data/route";
import tripInformation from "../data/trip-information";
import fullBooking from "../data/full-booking-list";
export const initialStateThree = {
  bookings: fullBooking,
  bookingAllowed: true,
  route: tripInformation.route.map(e => {
    e.count = 0;
    return e;
  }),
  captainLocation: {
    lat: null,
    lng: null
  },
  tripStatus: "NOT_STARTED",
  tripStartTime: null,
  arrivalStatus: "",
  currentStationId: null,
  fullDistance: null,
  baseFare: tripInformation.baseFare,
  startPoint: tripInformation.startPoint,
  endPoint: tripInformation.endPoint,
  driver: tripInformation.driver
};
