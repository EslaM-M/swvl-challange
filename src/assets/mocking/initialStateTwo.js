// import route from "../data/route";
import tripInformation from "../data/trip-information";
import bookings from "../data/booking";
export const initialState = {
  bookings: [],
  bookingAllowed: true,
  route: tripInformation.route,
  bookings: bookings,
  captainLocation: {
    lat: null,
    lng: null
  },
  tripStatus: "NOT_STARTED",
  tripStartTime: null,
  arrivalStatus: "",
  currentStationId: null,
  fullDistance: null,
  baseFare: tripInformation.baseFare,
  startPoint: tripInformation.startPoint,
  endPoint: tripInformation.endPoint,
  driver: tripInformation.driver
};
