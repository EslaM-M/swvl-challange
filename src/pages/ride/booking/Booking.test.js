import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Booking from "./index";
import { initialState } from "../../../assets/mocking/initialStateTwo";
describe("NewBooking Component", () => {
  let realUseContext;
  let useContextMock;
  // Setup mock
  beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
    useContextMock.mockReturnValue({
      state: initialState
    });
  });
  // Cleanup mock
  afterEach(() => {
    React.useContext = realUseContext;
  });

  test("Component snapshot render correctly", () => {
    const container = render(<Booking />);
    expect(container).toMatchSnapshot();
  });

  test("Component show the same number of booking in the state", async () => {
    const { getByText, baseElement } = render(<Booking />);
    const bookings = baseElement.querySelectorAll(".userBooking");
    expect(bookings.length).toBe(4);
  });
  test("Component show zero number of booking if there is no booking", async () => {

    initialState.bookings = [];
    useContextMock.mockReturnValue({
      state: initialState
    });

    const { getByText, baseElement } = render(<Booking />);

    const bookings = baseElement.querySelectorAll(".userBooking");
    const emptyCard = getByText("Add New Booking To Start Your Trip");
    expect(bookings.length).toBe(0);
    expect(emptyCard).toBeInTheDocument();
  });
});
