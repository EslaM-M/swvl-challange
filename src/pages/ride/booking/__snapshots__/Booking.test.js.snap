// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBooking Component Component snapshot render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="header"
        >
          Bookings
          <button
            class="btn btn-primary"
            type="button"
          >
            New Book
          </button>
        </div>
        <div
          class="container"
        >
          <div
            class="userBooking"
          >
            <div>
              <img
                alt="user-image"
                class="avatar"
                src="1.jpg"
              />
            </div>
            <div>
              <p
                data-testid="user-name"
              >
                Eslam Mostafa
              </p>
              <p>
                <label
                  data-testid="user-id"
                >
                  939
                </label>
                 
                <label
                  data-testid="booking-status"
                >
                  Canceled
                </label>
              </p>
            </div>
          </div>
          <div
            class="userBooking"
          >
            <div>
              <img
                alt="user-image"
                class="avatar"
                src="1.jpg"
              />
            </div>
            <div>
              <p
                data-testid="user-name"
              >
                Hossam Hassan
              </p>
              <p>
                <label
                  data-testid="user-id"
                >
                  935
                </label>
                 
                <label
                  data-testid="booking-status"
                >
                  Canceled
                </label>
              </p>
            </div>
          </div>
          <div
            class="userBooking"
          >
            <div>
              <img
                alt="user-image"
                class="avatar"
                src="1.jpg"
              />
            </div>
            <div>
              <p
                data-testid="user-name"
              >
                Hossam Hassan
              </p>
              <p>
                <label
                  data-testid="user-id"
                >
                  936
                </label>
                 
                <label
                  data-testid="booking-status"
                >
                  Compeleted
                </label>
              </p>
            </div>
          </div>
          <div
            class="userBooking"
          >
            <div>
              <img
                alt="user-image"
                class="avatar"
                src="1.jpg"
              />
            </div>
            <div>
              <p
                data-testid="user-name"
              >
                Hossam Hassan
              </p>
              <p>
                <label
                  data-testid="user-id"
                >
                  937
                </label>
                 
                <label
                  data-testid="booking-status"
                >
                  Compeleted
                </label>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="header"
      >
        Bookings
        <button
          class="btn btn-primary"
          type="button"
        >
          New Book
        </button>
      </div>
      <div
        class="container"
      >
        <div
          class="userBooking"
        >
          <div>
            <img
              alt="user-image"
              class="avatar"
              src="1.jpg"
            />
          </div>
          <div>
            <p
              data-testid="user-name"
            >
              Eslam Mostafa
            </p>
            <p>
              <label
                data-testid="user-id"
              >
                939
              </label>
               
              <label
                data-testid="booking-status"
              >
                Canceled
              </label>
            </p>
          </div>
        </div>
        <div
          class="userBooking"
        >
          <div>
            <img
              alt="user-image"
              class="avatar"
              src="1.jpg"
            />
          </div>
          <div>
            <p
              data-testid="user-name"
            >
              Hossam Hassan
            </p>
            <p>
              <label
                data-testid="user-id"
              >
                935
              </label>
               
              <label
                data-testid="booking-status"
              >
                Canceled
              </label>
            </p>
          </div>
        </div>
        <div
          class="userBooking"
        >
          <div>
            <img
              alt="user-image"
              class="avatar"
              src="1.jpg"
            />
          </div>
          <div>
            <p
              data-testid="user-name"
            >
              Hossam Hassan
            </p>
            <p>
              <label
                data-testid="user-id"
              >
                936
              </label>
               
              <label
                data-testid="booking-status"
              >
                Compeleted
              </label>
            </p>
          </div>
        </div>
        <div
          class="userBooking"
        >
          <div>
            <img
              alt="user-image"
              class="avatar"
              src="1.jpg"
            />
          </div>
          <div>
            <p
              data-testid="user-name"
            >
              Hossam Hassan
            </p>
            <p>
              <label
                data-testid="user-id"
              >
                937
              </label>
               
              <label
                data-testid="booking-status"
              >
                Compeleted
              </label>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
